### PR TITLE
Show events completion inside NameOf

### LINKED ISSUE #46472: VB.NET: IntelliSense doesn't show Events when using NameOf
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/1131227/vbnet-intellisense-doesnt-show-events-when-using-n.html)._

---
I have a class

```
Public Class Foo
  Event Bar()
  
Public Sub Baz()
    Dim Name1 = NameOf(Bar)
    Dim Name2 = NameOf(Me.Bar)
    Dim Name3 = NameOf(Foo.Bar)
  End Sub
End Class
```
This code works and in the variables are the right names/strings, but IntelliSense doesn't show the event name, when I write

```
Dim Name1 = NameOf(
```
Other members of a class are shown, but no events.

---
#### Feedback Bot on 7/30/2020, 10:50 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #46472

### CODE REVIEW

File: src/Workspaces/VisualBasic/Portable/Recommendations/VisualBasicRecommendationServiceRunner.vb [hunk 1]
```diff
@@ -341,11 +341,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Recommendations
...
 
         ''' <summary>
-        ''' In MemberAccessExpression Contexts, filter out event symbols, except inside AddRemoveHandler Statements
+        ''' In MemberAccessExpression Contexts, filter out event symbols (except for NameOf context), except inside AddRemoveHandler Statements
         ''' Also, filter out any implicitly declared members generated by event declaration or property declaration
         ''' </summary>
-        Private Shared Function FilterEventsAndGeneratedSymbols(node As MemberAccessExpressionSyntax, s As ISymbol) As Boolean
-            If s.Kind = SymbolKind.Event Then
+        Private Function FilterEventsAndGeneratedSymbols(node As MemberAccessExpressionSyntax, s As ISymbol) As Boolean
+            If s.Kind = SymbolKind.Event AndAlso Not _context.IsNameOfContext Then
```
Comment (Youssef1313) [inline src/Workspaces/VisualBasic/Portable/Recommendations/VisualBasicRecommendationServiceRunner.vb:348]: I was very specific here to the linked issue and allowed extended events to only NameOf.

Are there any other places where events should show in completion but they currently don't?

**Comment by CyrusNajmabadi** [comment]
LGTM.  please let me know when this passes.

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by Youssef1313** [comment]
@CyrusNajmabadi The build passed.

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Pull request merged**